include/shad/core/array.h

```diff
@@ -0,0 +1,214 @@
+  constexpr const_reference front() const { return ptr->front(); }
+
+  /// @brief the last element in the array.
+  /// @return a ::reference to the element in position N - 1.
+  constexpr reference back() { return ptr->back(); }
+
+  /// @brief the last element in the array.
+  /// @return a ::const_reference to the element in position N - 1.
+  constexpr const_reference back() const { return ptr->back(); }
+  /// @}
+
+  /// @defgroup Iterators
+  /// @{
+  /// @brief The iterator to the beginning of the sequence.
+  /// @return an ::iterator to the beginning of the sequence.
+  constexpr iterator begin() noexcept { return ptr->begin(); }
+
+  /// @brief The iterator to the beginning of the sequence.
+  /// @return a ::const_iterator to the beginning of the sequence.
+  constexpr const_iterator begin() const noexcept { return ptr->begin(); }
+
+  /// @brief The iterator to the end of the sequence.
+  /// @return an ::iterator to the end of the sequence.
+  constexpr iterator end() noexcept { return ptr->end(); }
+
+  /// @brief The iterator to the end of the sequence.
+  /// @return a ::const_iterator to the end of the sequence.
+  constexpr const_iterator end() const noexcept { return ptr->end(); }
+
+  /// @brief The iterator to the beginning of the sequence.
+  /// @return a ::const_iterator to the beginning of the sequence.
+  constexpr const_iterator cbegin() const noexcept { return ptr->cbegin(); }
+
+  /// @brief The iterator to the end of the sequence.
+  /// @return a ::const_iterator to the end of the sequence.
+  constexpr const_iterator cend() const noexcept { return ptr->cend(); }
+
+  // todo rbegin
+  // todo crbegin
+  // todo rend
+  // todo crend
+  /// @}
+
+  /// @defgroup Capacity
+  /// @{
+  /// @brief Empty test.
+  /// @return true if empty (N=0), and false otherwise.
+  constexpr bool empty() const noexcept { return ptr->empty(); }
+
+  /// @brief The size of the container.
+  /// @return the size of the container (N).
+  constexpr size_type size() const noexcept { return ptr->size(); }
+
+  /// @brief The maximum size of the container.
+  /// @return the maximum size of the container (N).
+  constexpr size_type max_size() const noexcept { return ptr->max_size(); }
+  /// @}
+
+  /// @defgroup Operations
+  /// @{
+  /// @brief Fill the array with an input value.
+  ///
+  /// @param v The input value used to fill the array.
+  void fill(const value_type &v) { ptr->fill(v); }
+
+  /// @brief Swap the content of two array.
+  ///
+  /// @param O The array to swap the content with.
+  void swap(array<T, N> &O) noexcept /* (std::is_nothrow_swappable_v<T>) */ {
+    ptr->swap(O);
+  }
+  /// @}
+
+  /// @defgroup Obsolete - todo
+  /// @{
+  auto get() { return ptr; }
+  /// @}
+
+ private:
+  std::shared_ptr<array_t> ptr = nullptr;
+};
+
+template <typename T, std::size_t N>
+bool operator==(const array<T, N> &LHS, const array<T, N> &RHS) {
+  return operator==(*LHS.get(), *RHS.get());
```

Comment:
Oh... I got it... friendship?

right now get is in the public API of the class.